include/swift/Frontend/Frontend.h

```diff
@@ -433,6 +449,57 @@ class CompilerInstance {
   /// Frees up the ASTContext and SILModule objects that this instance is
   /// holding on.
   void freeContextAndSIL();
+
+private:
+  /// Load stdlib & true if should continue, i.e. no error
+  bool loadStdlib();
+  ModuleDecl *importUnderlyingModule();
+  ModuleDecl *importBridgingHeader();
+
+  void
+  getImplicitlyImportedModules(SmallVectorImpl<ModuleDecl *> &importModules);
+
+public: // for static functions in Frontend.cpp
+  struct ImplicitImports {
+    SourceFile::ImplicitModuleImportKind kind;
+    ModuleDecl *objCModuleUnderlyingMixedFramework;
+    ModuleDecl *headerModule;
+    SmallVector<ModuleDecl *, 4> modules;
+
+    explicit ImplicitImports(CompilerInstance &compiler);
+  };
+
+private:
+  void createREPLFile(ImplicitImports &implicitImports);
+  std::unique_ptr<DelayedParsingCallbacks> computeDelayedParsingCallback();
+
+  void addMainFileToModule(ImplicitImports &implicitImports);
+
+  void parseAndCheckTypes(ImplicitImports &implicitImports);
+
+  void parseLibraryFile(unsigned BufferID, ImplicitImports &implicitImports,
+                        PersistentParserState &PersistentState,
+                        DelayedParsingCallbacks *DelayedParseCB);
+
+  /// Return true if had load error
+  bool
+  parsePartialModulesAndLibraryFiles(ImplicitImports &implicitImports,
+                                     PersistentParserState &PersistentState,
+                                     DelayedParsingCallbacks *DelayedParseCB);
+
+  void checkTypesWhileParsingMain(PersistentParserState &PersistentState,
+                                  DelayedParsingCallbacks *DelayedParseCB);
+
+  OptionSet<TypeCheckingFlags> computeTypeCheckingOptions();
+
+  void forEachFileToTypeCheck(const std::function<void(SourceFile &)> &fn);
```

Comment:
This one I'm more happy with. :-) `llvm::function_ref` again, though.

Reply:
Good catch.